service/controller/rbac/resource/fluxauth/create.go

```diff
@@ -0,0 +1,242 @@
+		roleBinding.Subjects = append(roleBinding.Subjects,
+			rbacv1.Subject{
+				Kind:      "ServiceAccount",
+				Name:      serviceAccount,
+				Namespace: fluxNamespace,
+			},
+		)
+	}
+
+	if err := r.createOrUpdateRoleBinding(ctx, ns, roleBinding); err != nil {
+		return microerror.Mask(err)
+	}
+
+	// create a RoleBinding allowing *some* ServiceAccounts in flux-system to
+	// reconcile (read, write) Flux CRs in org namespace
+	roleBinding = &rbacv1.RoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "RoleBinding",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pkgkey.FluxReconcilerRoleBindingName,
+			Labels: map[string]string{
+				label.ManagedBy: project.Name(),
+			},
+			Namespace: ns.Name,
+		},
+		Subjects: []rbacv1.Subject{},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     pkgkey.ClusterAdminClusterRoleName,
+		},
+	}
+
+	for _, serviceAccount := range reconcilerServiceAccounts {
+		roleBinding.Subjects = append(roleBinding.Subjects,
+			rbacv1.Subject{
+				Kind:      "ServiceAccount",
+				Name:      serviceAccount,
+				Namespace: fluxNamespace,
+			},
+		)
+	}
+
+	if err := r.createOrUpdateRoleBinding(ctx, ns, roleBinding); err != nil {
+		return microerror.Mask(err)
+	}
+	return nil
+}
+
+func (r *Resource) createOrUpdateRoleBinding(ctx context.Context, ns corev1.Namespace, roleBinding *rbacv1.RoleBinding) error {
+	existingRoleBinding, err := r.k8sClient.RbacV1().RoleBindings(ns.Name).Get(ctx, roleBinding.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating rolebinding %#q", roleBinding.Name))
+
+		_, err := r.k8sClient.RbacV1().RoleBindings(ns.Name).Create(ctx, roleBinding, metav1.CreateOptions{})
+		if apierrors.IsAlreadyExists(err) {
+			// do nothing
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("rolebinding %#q has been created", roleBinding.Name))
+
+	} else if err != nil {
+		return microerror.Mask(err)
+	} else if needsUpdate(roleBinding, existingRoleBinding) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating role binding %#q", roleBinding.Name))
+		_, err := r.k8sClient.RbacV1().RoleBindings(ns.Name).Update(ctx, roleBinding, metav1.UpdateOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("role binding %#q has been updated", roleBinding.Name))
+
+	} else {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("rolebinding %#q already exists", roleBinding.Name))
+	}
+
+	return nil
+}
+
+func needsUpdate(desiredRoleBinding, existingRoleBinding *rbacv1.RoleBinding) bool {
+	if len(existingRoleBinding.Subjects) < 1 {
+		return true
+	}
+
+	if desiredRoleBinding.Subjects[0].Name != existingRoleBinding.Subjects[0].Name {
```

Comment:
Dumb question but what happens if there is more than one subject?

Would reflect.DeepEqual work instead?